[Feature] Implement AI-powered product descriptions using Gemini

ViralsSolguruz opened the pull request:
# Pull Request

## Description
- Adds `google_generative_ai` dependency for Gemini integration.
- Implements `GeminiService` to handle API interactions.
- Integrates a new "AI-Powered Description" section on the product detail screen, which allows users to generate a description.
- Includes a loading shimmer, generated content view, and error handling with a retry option.
- Adds personalization to the AI prompt based on the user's mock order history.
- Implements the necessary Bloc logic, states, events, use cases, repositories, and data sources for the feature.
- Sets up dependency injection for all new AI-related components.
- Adds new localization strings for the feature.

- Following are the screenshots of AI-generated product details:

<img width="200" alt="android_product_1" src="https://github.com/user-attachments/assets/c8ed3fc0-0f11-4079-bd25-57752ca77b34" />
<img width="200" alt="android_product_2" src="https://github.com/user-attachments/assets/99073215-a86f-4fdd-b1c1-e8195e2f174a" />
<img width="200" alt="IMG_0272" src="https://github.com/user-attachments/assets/795fff73-3722-431d-8723-fd5735c1a09e" />
<img width="200" height="2532" alt="IMG_0273" src="https://github.com/user-attachments/assets/a2e72e4a-0a62-4227-b465-1cdf3d4eab14" />


## Type of Change
<!-- Please check the appropriate options -->
- [ ] Bug fix
- [x] New feature
- [ ] Code refactoring
- [ ] Test cases
- [ ] Documentation update
- [ ] UI/UX improvement
- [ ] Performance improvement
- [ ] Build configuration change
- [ ] Other (please describe):

## Screenshots (if applicable)
<!-- Add screenshots here if UI changes were made (Skip if golden tests are added) -->

## Checklist
- [x] Before requesting for review, I have self reviewed all the changes in this PR
- [x] My code follows the project's coding style guidelines
- [x] I have added any necessary tests
- [x] New and existing tests pass locally with my changes
- [ ] Github pipeline passes with my changes

## Additional Notes
<!-- Any additional information that reviewers should know -->

ViralsSolguruz committed:
feat: Implement AI-powered product descriptions using Gemini

This commit introduces a new feature to generate AI-powered product descriptions on the product detail screen.

- Adds `google_generative_ai` dependency for Gemini integration.
- Implements `GeminiService` to handle API interactions.
- Integrates a new "AI-Powered Description" section on the product detail screen, which allows users to generate a description.
- Includes a loading shimmer, generated content view, and error handling with a retry option.
- Adds personalization to the AI prompt based on the user's mock order history.
- Implements the necessary Bloc logic, states, events, use cases, repositories, and data sources for the feature.
- Sets up dependency injection for all new AI-related components.
- Adds new localization strings for the feature.

ViralsSolguruz committed:
ref: add AI Product Description Implementation Guide

This commit introduces a comprehensive guide for implementing and integrating AI-powered product descriptions using Google's Gemini AI.

The guide covers:
- Overview and prerequisites
- Architecture and directory structure
- Detailed setup and implementation instructions
- Usage, customization, and troubleshooting
- Best practices for caching, error handling, and security

mihirjsolguruz commented on lib/presentation/product_detail/data/models/ai_product_description_model.dart:
```diff
@@ -0,0 +1,63 @@
+import 'package:skelter/presentation/product_detail/domain/entities/ai_product_description.dart';
+import 'package:skelter/utils/typedef.dart';
+
+class AiProductDescriptionModel extends AiProductDescription {
```
Please change the spelling of Ai to AI across the PR

ViralsSolguruz replied:
Done

mihirjsolguruz commented on lib/presentation/product_detail/widgets/ai_product_description.dart:
```diff
@@ -0,0 +1,308 @@
+            else if (hasAiDescription)
+              _buildAiDescriptionContent(context, state)
+            else
+              _buildGenerateButton(context),
```
please move this widgets in a separate file and as a separate widget class

ViralsSolguruz replied:
Done

mihirjsolguruz commented on pubspec.yaml:
```diff
@@ -82,6 +82,7 @@ dependencies:
   pdf: ^3.11.1
   printing: ^5.14.0
   timezone: ^0.10.1
+  google_generative_ai: ^0.4.6
```
this lib is deprecated. Can we change it with Firebase AI?
<img width="892" height="665" alt="Image" src="https://github.com/user-attachments/assets/7fda9d07-1600-4da9-ba58-a59e7f07003a" />

ViralsSolguruz replied:
Done

mihirjsolguruz commented on AI_PRODUCT_DESCRIPTION_GUIDE.md:
if purpose of this file is guide, then I think we should move it into Skelter Docs

ViralsSolguruz replied:
Done

mihirjsolguruz commented:
@ViralsSolguruz please remove unnecessary comments from PR

ViralsSolguruz committed:
ref: Rename Ai-related classes and variables to AI

Standardizes the casing for "AI" across various files related to the AI product description feature for consistency. This includes renaming classes, variables, and state/event names.

ViralsSolguruz committed:
ref: Restructure AI product description widget

Refactor the `AIProductDescription` widget by extracting its components into smaller, more manageable widgets.

This change introduces the following new widgets:
- `AiDescriptionHeader`
- `AiDescriptionShimmer`
- `AiDescriptionContent`
- `AiDescriptionErrorWidget`
- `AiDescriptionGenerateButton`

The main `AIProductDescription` widget is simplified to orchestrate these new components based on the bloc state, improving readability and maintainability.

ViralsSolguruz committed:
ref: migrate from Google AI SDK to Firebase AI

This commit refactors the `GeminiService` to use the `firebase_ai` package instead of the `google_generative_ai` package.

Key changes include:
*   Replacing the `google_generative_ai` dependency with `firebase_ai` in `pubspec.yaml`.
*   Removing the manual API key management logic (`_apiKey`, `dotenv`) as Firebase AI handles authentication automatically.
*   Updating the service initialization to use `FirebaseAI.googleAI().generativeModel()` for creating both text and vision models.
*   Adjusting comments and debug messages to reflect the use of Firebase AI.

ViralsSolguruz committed:
ref: remove AI product description guide

ViralsSolguruz committed:
ref: remove redundant comments

ViralsSolguruz committed:
refactor: rename AI description widgets

Renames widgets related to the AI product description to follow the 'AI' naming convention instead of 'Ai'. This affects the following widgets:
- `AIDescriptionContent`
- `AIDescriptionErrorWidget`
- `AIDescriptionGenerateButton`
- `AIDescriptionHeader`
- `AIDescriptionShimmer`

ViralsSolguruz commented:
> @ViralsSolguruz please remove unnecessary comments from PR

Done

mihirjsolguruz commented on pubspec.yaml:
```diff
@@ -82,6 +82,9 @@ dependencies:
   pdf: ^3.11.1
   printing: ^5.14.0
   timezone: ^0.10.1
+  # With:
```
remove this unnecessary comment
```suggestion
```

ViralsSolguruz replied:
Done

mihirjsolguruz commented on lib/services/ai/gemini_constants.dart:
```diff
@@ -0,0 +1,102 @@
+  static const double temperature = 0.7;
+
+  /// Maximum tokens in the response
+  static const int maxOutputTokens = 1536/**/;
```
```suggestion
  static const int maxOutputTokens = 1536;
```

ViralsSolguruz replied:
Done

ViralsSolguruz committed:
ref: Remove unnecessary comments
